### PR TITLE
feat(quote): add support for fee on transfer token

### PR DESCRIPTION
**Pull request by mul53**
# Description

add support for fee on transfer tokens

### CODE REVIEW

**Review by alonweinstein — APPROVED** [review]

File: src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts [hunk 1]
```diff
@@ -46,6 +47,7 @@ import {
     getMaxQuoteSlippageRate,
     getTransformerNonces,
     isDirectSwapCompatible,
+    isFeeTransferTokenCompitable as isFeeTransferTokenFillCompatible,
```
Comment (darkpaladi) [inline src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts]: Maybe we can import this with the same name

File: src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts [hunk 2]
```diff
@@ -492,4 +504,33 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumer {
                 .getABIEncodedTransactionData();
         }
     }
+
+    private encodeUniswapV2FillCalldata(quote: SwapQuote, opts: ExchangeProxyContractOpts & { takerAddress?: string }): string {
```
Comment (darkpaladi) [inline src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts]: I saw that you created a method for swapExactTokensForTokensSupportingFeeOnTransferTokens , what about for:
- swapExactETHForTokensSupportingFeeOnTransferTokens  --> FUSE (native) to fee token
- swapExactTokensForETHSupportingFeeOnTransferTokens  --> Fee token to FUSE (native)

Do we need to add support for those cases?

File: src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts (hunk 1 above)
Comment (mul53) [reply src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts]: yes correct

File: src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts (hunk 2 above)
Comment (alonweinstein) [reply src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts]: @mul53 ?

Comment (mul53) [reply src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts]: good catch @darkpaladi, the PR now should be okay

**Review by darkpaladi — APPROVED** [review]

**Pull request merged**